EIPS/eip-5131.md

```diff
@@ -0,0 +1,294 @@
+
+### Example 1: Event Access
+An event requires attendees to prove ownership of an NFT to gain access. The attendee in question only has their phone available to perform a signing operation upon the presentation of a challenge. The phone is considered the attendees 'hot wallet' which is linked to auth.wilkins.eth.
+The NFT is owned by the Ethereum address that corresponds to wilkins.eth, which is controlled by the attendees' cold storage device (Ledger, Trezor etc). The attendee does not have this device present at the event.
+The attendee is able to prove ownership of the NFT by signing with the 'hot wallet' key.
+
+This is made possible because the attendee completed the necessary upfront authorization steps with the cold storage device to allow the 'hot wallet' to be used for this operation. 
+
+### Example 2: Multiple Devices
+I have multiple devices and wish to sign with any of them for convenience.  In this situation:
+ - My phone has mobile metamask hot wallet, and I can add it as an auth account by setting auth1.wilkins.eth to that wallet (and the corresponding reverse record). 
+ - My tablet has the same and can be set to auth2.wilkins.eth, etc. 
+
+Lost Access to Signing Key/Device: 
+In the event that a user losses access to the signing key of an `authAddress` they can simply delete the corresponding ENS record. An attractive side effect of this is that there is no need to import/reimport seed phrases. 
+In the unfortunate event that you lose access to the signing key of the `mainAddress`, users will lose access to that top-level domain and all related subdomains. As always, this private key data should be treated as highly sensitive. 
+
+
+## Reference Implementation
+
+### Client/Server Side
+In typescript, the validation function, using ethers.js would be as follows:
+```
+export interface LinkedAddress {
+  ens: string,
+  address: string,
+}
+
+async function getLinkedAddress(provider: ethers.providers.Provider, address: string): Promise<LinkedAddress | null> {
+  const addressENS = await provider.lookupAddress(address);
+  if (!addressENS) return null;
+
+  const authMatch = addressENS.match(/^(auth[0-9A-za-z]*)\.(.*)/);
+  if (!authMatch) return null;
+
+  const linkedENS = authMatch[2];
+  const linkedAddress = await provider.resolveName(linkedENS);
+
+  if (!linkedAddress) return null;
+
+  return {
+    ens: linkedENS,
+    address: linkedAddress
+  };
+}
+```
+
+### Contract side
+
+### With a secure server
+If your application operates a secure backend server, you could run the client/server code above, then use the result in conjunction with specs like [EIP-1271](./eip-1271.md) : `Standard Signature Validation Method for Contracts` for a cheap and secure way to validate that the the message signer is indeed authenticated for the main address.
+
+#### Without a secure server (web client only)
```

Comment:
I have trouble understanding what is meant by `secure server`. Does this refer to the node provider? Otherwise, I could run the above js code also on a web client.